### PR TITLE
add new data source of api gateway groups

### PR DESCRIPTION
**Pull request by geeklenny**
TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudApigateway -timeout=300m
=== RUN   TestAccAlicloudApigatewayGroupsDataSource_basic
--- PASS: TestAccAlicloudApigatewayGroupsDataSource_basic (6.46s)
=== RUN   TestAccAlicloudApigatewayGroup_importBasic
--- PASS: TestAccAlicloudApigatewayGroup_importBasic (5.04s)
=== RUN   TestAccAlicloudApigatewayGroup_basic
919653d77d844fa5b0ed664dc519c9b4
--- PASS: TestAccAlicloudApigatewayGroup_basic (20.83s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     32.395s

### CODE REVIEW

File: alicloud/data_source_alicloud_api_gateway_groups.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+				},
+			},
+		},
+	}
+}
+func dataSourceAlicloudApigatewayGroupsRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AliyunClient).cloudapiconn
+
+	args := cloudapi.CreateDescribeApiGroupsRequest()
+	args.RegionId = string(getRegion(d, meta))
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_api_gateway_groups.go]: Please use `args.RegionId = getRegionId(d, meta)

File: alicloud/data_source_alicloud_api_gateway_groups.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		resp, err := conn.DescribeApiGroups(args)
+		if err != nil {
+			return err
+		}
+
+		if resp == nil {
+			break
+		}
+
+		allGroups = resp.ApiGroupAttributes.ApiGroupAttribute
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_api_gateway_groups.go]: I think line 99 should be `allGroups = append(allGroups, resp.ApiGroupAttributes.ApiGroupAttribute...)`

File: examples/api-gateway/main.tf [hunk 3]
```diff
@@ -1,9 +1,29 @@
-resource "alicloud_api_gateway_group" "apiGatewayGroup" {
```
Comment (xiaozhu36) [inline examples/api-gateway/main.tf]: I think there is no need to modify this example

File: website/docs/d/api_gateway_groups.html.markdown [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+output "first_group_id" {
+  value = "${data.alicloud_api_gateway_groups.data_apigatway.groups.0.id}"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+* `output_file` - (Optional) File name where to save data source results (after running `terraform plan`).
```
Comment (xiaozhu36) [inline website/docs/d/api_gateway_groups.html.markdown]: Missing parametes

Comment (geeklenny) [reply website/docs/d/api_gateway_groups.html.markdown]: all resolved

**Comment by geeklenny** [comment]
all issues resolved

**Pull request merged**